### ISSUE TITLE
Dask Fallback and ACME dependency

### ISSUE THREAD

**Issue by tensionhead** [esi-neuroscience/syncopy#371]
## Current status
Right now for the [2022.08](https://github.com/esi-neuroscience/syncopy/releases/tag/v2022.08) release, we have ACME as hard dependency. Before that release acme only was an optional dep, the reasoning to include it now as a hard dep was that Syncopy without any parallel processing capabilities is not that attractive. However, that makes versioning much harder due to hard version pinning in acme.. as we learned the hard way: acme 2022.7 depends on `python <3.9,>=3.7 ` which effectively limitted the supported Python versions of syncopy to exactly this.

## Idea
Return to acme being an optional dependency for the next release, but offer some limited parallel processing in Syncopy by using Dasks simple [Futures API](https://examples.dask.org/applications/embarrassingly-parallel.html). All of Syncopy's parallel computing (aka `ComputationalRoutine`s) needs  are embarassingly parallel, hence in principle that is very straightforward with Dask:
```python
from dask.distributed import Client, LocalCluster, get_client
def costly_func(par):
    ...
parameters = [1, 2, 3]
# use Dask client
client = Client()
# simple mapping, already triggers computation
futures = client.map(costly_func, parameters)
# waits till the last worker has finished
results = client.gather(futures)
```
However for syncopy, we don't really have `results` as the CRs write everything nicely into hdf5 and take care of collecting the results. 
For use on a single machine, the default `LocalCluster` already gives a huge speedup basically for free, as there is no special setup needed. Overall having standalone parallel computing capabilites could give more freedom for Syncopy deployment, by leaving the `client` setup to the respective user/IT department.

**Comment by dfsp-spirit** [comment]
Ah, acme was the reason for that Python version requirement, right.

The LocalCluster sounds like a great thing to support. It won't work with large trials on low-memory machines, but that's how it is.

On a side note, we should also encourage the acme developers to get rid of that Python requirement.

**Comment by pantaray** [comment]
Hi! The current dev version of ACME works with Python < 3.11 and dask < 2022.11. It should be out within the next two or three weeks (people are currently test-driving it). It also includes functionality to collect results via HDF5 `ExternalLinks` and virtual datasets. If wanted, some of the housekeeping machinery currently sitting in `computational_routine.py` (e.g., parts of `preallocate_output`)  could now be done directly inside ACME which might lighten the HDF5 administration in Syncopy. If that sounds interesting to you guys, I'd put together a PR

**Comment by tensionhead** [comment]
Hi, ok great to hear that the new acme will be free of hard Python version constraints! I think the way the CRs (and the critical `process_io` decorator) are set up right now gives us exactly the freedom to more or less make drop-in fallbacks for ACME's `ParallelMap` via using pure dask in a way I outlined above. So from my point of view, no need to further re-build CR functionality inside ACME.

**Comment by pantaray** [comment]
OK got, then I'll ping you as soon as the next ACME version is ready to roll.

**Comment by tensionhead** [comment]
Ok, so I reverted back to acme being an optional dependency in a feature branch.. but for this to work, we have to keep at least the conda requirements in sync across projects. @pantaray do you know of any tools which could help with this? At the moment its manually comparing both environment `.yml`'s then fire up conda and hope for the best :sweat_smile:

**Comment by pantaray** [comment]
Hi! I'm actually stumbling upon the very same problem prepping ACME's upcoming release. Unfortunately, off the top of my head I'm not aware of any tool that automates that. Maybe we can do some digging in the scikit repos? They must've stumbled upon the very same problem...

**Comment by tensionhead** [comment]
Ok I see.. so the current most up to date Syncopy dependencies are in `pyproject.toml` of the branch of PR #373 However the `syncopy.yml` in that very same branch tried to adhere to acme's 2022.8 dependencies.. so that is not something to look for guidance at the moment. Maybe I could try to `pip install -e` from ACME's dev branch, having the new updated dependencies you described above into an updated syncopy conda environemnt?

**Comment by pantaray** [comment]
Yupp, that's the way I'm currently testing syncopy/acme as well. I did not install the current pip versions of the packages but cloned the repos, switched to the respective `dev` branches and used `pip install -e .` on both of them

**Comment by tensionhead** [comment]
Got merged with #373
